roofit/roostats/inc/RooStats/RooStatsUtils.h

```diff
@@ -89,10 +85,9 @@ namespace RooStats {
    /// utility function to set all variable constant in a collection
    /// (from G. Petrucciani)
    inline bool SetAllConstant(const RooAbsCollection &coll, bool constant = true) {
-      bool changed = false;
-      RooLinkedListIter iter = coll.iterator();
-      for (RooAbsArg *a = (RooAbsArg *) iter.Next(); a != 0; a = (RooAbsArg *) iter.Next()) {
-         RooRealVar *v = dynamic_cast<RooRealVar *>(a);
+      bool changed = false; 
+      for (auto *v : static_range_cast<RooRealVar *>(coll)) {
+         //RooRealVar *v = dynamic_cast<RooRealVar *>(a);
```

Comment:
```suggestion
      for (auto *v : dynamic_range_cast<RooRealVar *>(coll)) {
```
Has to be dynamic cast because the result is checked for `nullptr` in the `if` below.